### PR TITLE
Update optimistic import rules

### LINKED ISSUE #5105: [Merge] Optimistically import any post-merge block
### Description
Update optimistic rules as per https://github.com/ethereum/consensus-specs/pull/2844#pullrequestreview-899202643

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Can now import a block optimistically if it's parent has a non-default payload or if it is sufficiently old.
Also moves the logic for determining if optimistic import is allowed into the spec module since it's a function from the spec itself.

## Fixed Issue(s)
fixes #5105 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java [hunk 1]
```diff
@@ -196,6 +200,68 @@ void onTick_shouldExitImmediatelyWhenCurrentTimeIsBeforeStoreTime() {
...
+  @Test
+  void canOptimisticallyImport_shouldBeTrueWhenParentBlockHasRealPayload() {
+    final SignedBeaconBlock blockToImport = dataStructureUtil.randomSignedBeaconBlock(11);
+    final ReadOnlyStore store =
+        mockStore(15, blockToImport.getParentRoot(), blockToImport.getRoot());
+    assertThat(forkChoiceUtil.canOptimisticallyImport(store, blockToImport)).isTrue();
+  }
+
+  @Test
+  void isOptimisticSyncPossible_shouldBeFalseWhenBlockIsMergeButNotOldEnough() {
```
Comment (tbenr) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: nit: rename to `canOptimisticallyImport_`

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java [hunk 2]
```diff
@@ -196,6 +200,68 @@ void onTick_shouldExitImmediatelyWhenCurrentTimeIsBeforeStoreTime() {
...
+    final int blockSlot = 11;
+    final SignedBeaconBlock blockToImport = dataStructureUtil.randomSignedBeaconBlock(blockSlot);
+    final ReadOnlyStore store =
+        mockStore(blockSlot + getSafeSyncDistance() - 1, blockToImport.getRoot());
+
+    assertThat(forkChoiceUtil.canOptimisticallyImport(store, blockToImport)).isFalse();
+  }
+
+  @Test
+  void isOptimisticSyncPossible_shouldBeTrueWhenBlockIsMergeAndIsOldEnough() {
```
Comment (tbenr) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: nit: rename to `canOptimisticallyImport_`

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtil.java [hunk 3]
```diff
@@ -439,4 +440,33 @@ private boolean hasAncestorAtSlot(
...
+        .plus(maybeConfig.get().getSafeSlotsToImportOptimistically())
+        .isLessThanOrEqualTo(getCurrentSlot(store));
+  }
+
+  private boolean isExecutionBlock(final ReadOnlyStore store, final Bytes32 blockRoot) {
+    final Optional<Bytes32> parentExecutionRoot =
+        store.getForkChoiceStrategy().executionBlockHash(blockRoot);
+    final boolean isExecutionBlock =
+        parentExecutionRoot.isPresent() && !parentExecutionRoot.get().isZero();
+    return isExecutionBlock;
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtil.java]: nit: the local boolean `isExecutionBlock` seems a bit redundant to me. The method itself has that name and should be enough to understand the objective.

**Review by tbenr — APPROVED** [review]
LGTM with some nits

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtil.java (hunk 3 above)
Comment (ajsutton) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtil.java]: IntelliJ agrees. :)

**Pull request merged**